### PR TITLE
[#171800318] preValidate with acceptedClockSkewMs

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/utils/__tests__/saml.test.ts [hunk 1]
```diff
@@ -83,29 +92,103 @@ describe("preValidateResponse", () => {
     await asyncExpectOnCallback(mockCallback);
   });
 
-  it("should preValidate succeded with a valid CIE saml Response", async () => {
-    const mockBody = "MOCKED BODY";
+  it("should preValidate succeded if timers are desyncronized and acceptedClockSkewMs is disabled", async () => {
```
Comment (gunzip) [inline src/utils/__tests__/saml.test.ts]:
```suggestion
  it("should preValidate succeed if timers are desynchronized and acceptedClockSkewMs is disabled", async () => {
```

File: src/utils/__tests__/saml.test.ts [hunk 2]
```diff
@@ -83,29 +92,103 @@ describe("preValidateResponse", () => {
...
+      { ...samlConfig, acceptedClockSkewMs: -1 },
       mockBody,
       mockRedisCacheProvider,
       mockCallback
     );
     expect(mockGetXmlFromSamlResponse).toBeCalledWith(mockBody);
     await asyncExpectOnCallback(mockCallback);
   });
+
+  it("should preValidate succeded if timers desync is lesser than acceptedClockSkewMs", async () => {
```
Comment (gunzip) [inline src/utils/__tests__/saml.test.ts]:
```suggestion
  it("should preValidate succeed if timers desync is less than acceptedClockSkewMs", async () => {
```

File: src/utils/saml.ts [hunk 3]
```diff
@@ -554,7 +554,11 @@ const mainAttributeValidation = (
     .chain(IssueInstant => utcStringToDate(IssueInstant, "IssueInstant"))
     .chain(
       fromPredicate(
-        _ => !hasStrictValidation || _.getTime() < Date.now(),
+        _ =>
+          _.getTime() <
```
Comment (gunzip) [inline src/utils/saml.ts]: can we move this computation at caller level (at the same level of hasStrictValidation) so we can reuse it seamlessy between inner methods ? ie.

```
const acceptedClockSkewMs = (samlConfig.acceptedClockSkewMs || 0) === -1
            ? Infinity
            : Date.now() + samlConfig.acceptedClockSkewMs);
```

Comment (BurnedMarshal) [reply src/utils/saml.ts]: In two cases the value is `Infinity` (NotBefore and IssueInstant) and on another case is `-Infinity` (NotOnOrAfter). How we can handle these scenarios?

Comment (gunzip) [reply src/utils/saml.ts]: sorry I didn't see this, it's ok as is then

**Pull request merged**